lib/rspec/support/differ.rb

```diff
@@ -178,12 +181,7 @@ def object_to_string(object)
         object = @object_preparer.call(object)
         case object
         when Hash
-          object.keys.sort_by { |k| k.to_s }.map do |key|
-            pp_key   = PP.singleline_pp(key, "")
-            pp_value = PP.singleline_pp(object[key], "")
-
-            "#{pp_key} => #{pp_value},"
-          end.join("\n")
+          ObjectInspector.inspect(object)
```

Comment:
Are there other situations where we'd want the `ObjectInspector` to be used?  Maybe for arrays?  I wonder if we can do away with `pp` entirely?